### ISSUE TITLE
Flutter Material Icons is not showing correctly 

### ISSUE THREAD

**Issue by minayoussefrofael** [flutter/flutter#108746]
Flutter material icons are not showing correctly as per the below photo, however other packages icons like awesomefonts and CupertinoIcons show without any issues

<img src="https://user-images.githubusercontent.com/108657251/182218922-380fafc3-b063-42c2-9d06-2852a2c60b39.png" width="300" />

<details>
<summary>sample</summary>

```dart
backgroundColor: Colors.grey[500],
                      elevation: 10,
                      child: const Icon(
                        Icons.post_add,
                        color: Color(0xFF0C0563),
                      ),
                    ),
                  ),
                ),
                Positioned(
                    left: widget.size.width * 0.05,
                    bottom: 12,
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        SizedBox(
                          height: 30,
                          width: 40,
                          child: IconButton(
                              onPressed: () {
                                Navigator.of(context)
                                    .popAndPushNamed('Homepage');
                              },
                              icon: const Icon(
                                Icons.home,
                                color: Colors.white,
                              )),
                        ),
                        const Text(
                          'Home',
                          style: TextStyle(color: Colors.white, fontSize: 11),
                        ),
                      ],
                    )),
                Positioned(
                    left: widget.size.width * 0.23,
                    bottom: 28,
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        SizedBox(
                          height: 30,
                          width: 40,
                          child: IconButton(
                              onPressed: () {},
                              icon: const Icon(
                                Icons.notifications,
                                color: Colors.white,
                              )),
                        ),
                        const Text(
                          'Notifications',
                          style: TextStyle(color: Colors.white, fontSize: 11),
                        ),
                      ],
                    )),
                Positioned(
                    left: widget.size.width * 0.65,
                    bottom: 28,
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        SizedBox(
                          height: 30,
                          width: 40,
                          child: IconButton(
                              onPressed: () {
                                Navigator.of(context)
                                    .popAndPushNamed('PeopleTab');
                              },
                              icon: const Icon(
                                Icons.people_alt,
                                color: Colors.white,
                              )),
                        ),
                        const Text(
                          'Colleagues',
                          style: TextStyle(color: Colors.white, fontSize: 11),
                        ),
                      ],
                    )),
                Positioned(
                    left: widget.size.width * 0.86,
                    bottom: 12,
                    child: Column(
                      mainAxisAlignment: MainAxisAlignment.center,
                      children: [
                        SizedBox(
                          height: 30,
                          width: 40,
                          child: IconButton(
                              onPressed: () {},
                              icon: const Icon(
                                Icons.chat,
                                color: Colors.white,
                              )),
                        ),
```
</details>

also in pubspec.yaml this is set to true   `uses-material-design: true`
and tried to make flutter clean and pub get with no effect at all.
and below is the flutter doctor summary 

<details>
<summary>flutter doctor</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [Version 10.0.19043.1826], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.6)   
[√] Android Studio (version 2021.2)
[√] VS Code (version 1.69.2)
[√] Connected device (4 available)
[√] HTTP Host Availability

• No issues found!
```
</details>

**Comment by danagbemava-nc** [comment]
Hi @minayoussefrofael, the code sample you have provided is not actionable for us. Please provide a complete minimal reproducible code sample so that we investigate this. 

Do you also experience this issue with the default flutter create app? 

Please also provide your pubspec.yaml

Thank you

**Comment by michaelprimez** [comment]
Hi all,

Unfortunately, I am having the same problem. 
Below, I past a minimum code that, at least in my case, doesn't present the correct icons:

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Test Icons'),
        ),
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Column(
                mainAxisAlignment: MainAxisAlignment.spaceAround,
                children: const <Widget>[
                  Icon(
                    Icons.favorite,
                    color: Colors.pink,
                    size: 24.0,
                    semanticLabel: 'Text to announce in accessibility modes',
                  ),
                  Icon(
                    Icons.audiotrack,
                    color: Colors.green,
                    size: 30.0,
                  ),
                  Icon(
                    Icons.beach_access,
                    color: Colors.blue,
                    size: 36.0,
                  ),
                  Icon(
                    Icons.home,
                  ),
                ],
              ),
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceAround,
                children: const <Widget>[
                  Icon(
                    Icons.people,
                    size: 24.0,
                  ),
                  Icon(
                    Icons.password,
                    size: 30.0,
                  ),
                  Icon(
                    Icons.email,
                    size: 36.0,
                  ),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```

flutter doctor -v
```
flutter doctor -v
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-GB)
    • Flutter version 3.0.5 at /Users/michael/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (3 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/michael/Library/Android/sdk
    • Platform android-32, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 67.1.4
    • Dart plugin version 221.5591.58

[✓] VS Code (version 1.69.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • MICHAIL KESKINIDIS’s iPhone (mobile) • 00008101-000025242651001E • ios            • iOS 15.5 19F77
    • macOS (desktop)                      • macos                     • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)                         • chrome                    • web-javascript • Google Chrome 103.0.5060.134

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
flutter upgrade 
Flutter is already up to date on channel stable
Flutter 3.0.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f1875d570e (3 weeks ago) • 2022-07-13 11:24:16 -0700
Engine • revision e85ea0e79c
Tools • Dart 2.17.6 • DevTools 2.12.2

```

Screenshot: iPhone 12 Pro, iOS: 15.5, 
![thumbnail_Screenshot 2022-08-02 at 19 47 17](https://user-images.githubusercontent.com/7694190/182441504-dc382b43-1b1c-4e23-ab10-0bf6b85f3d3b.jpg)

Screenshot: Web
<img width="843" alt="Screenshot 2022-08-02 at 19 56 32" src="https://user-images.githubusercontent.com/7694190/182442283-8f538bcb-ca57-4b82-959f-43e21a989380.png">

**Comment by danagbemava-nc** [comment]
<img width="1402" alt="Screenshot 2022-08-03 at 06 24 02" src="https://user-images.githubusercontent.com/88313112/182538581-f211c427-28cb-429c-b799-c67387ed7f2a.png">

Hi @michaelprimez, I can't reproduce this bug with the code sample that you have provided. 
When did you start experiencing this bug?

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-GB)
    • Flutter version 3.0.5 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (3 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8609683/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.69.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (4 available)
    • M2007J20CG (mobile) • 5dd3be00                  • android-arm64  • Android 11 (API 30)
    • Nexus (mobile)      • 00008020-001875E83A38002E • ios            • iOS 15.6 19G71
    • macOS (desktop)     • macos                     • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)        • chrome                    • web-javascript • Google Chrome 103.0.5060.134
    ! Error: Nexus is busy: Fetching debug symbols for Nexus. Xcode will continue when Nexus is finished. (code -10)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by michaelprimez** [comment]
Hi @danagbemava-nc,

Unfortunately, is hard to tell with certainty.
I did 2 things before this problem appeared:
1. flutter upgrade and
2. I've updated the below dependancies 
-  firebase_core: ^1.10.6 ->  ^1.20.0
-  firebase_auth: ^3.3.13 -> ^3.5.1
-  firebase_storage: ^10.2.4 -> ^10.3.3
-  cloud_firestore: ^3.1.5 -> ^3.4.1
-  cloud_functions: ^3.2.4 -> ^3.3.3
-  firebase_crashlytics: ^2.8.4 -> ^2.8.6
-  firebase_analytics: ^9.2.0 -> ^9.3.0
-   google_fonts: ^2.2.0 -> ^3.0.1
-   webview_flutter: ^3.0.0 - > ^3.0.4
-  html_editor_enhanced: ^2.4.0+1 -> ^2.5.0
-  responsive_framework: ^0.1.7 -> ^0.2.0

I've added the below dependancies
-  social_auth_buttons: ^2.0.0
-  dartz: ^0.10.1
-  image_editor: ^1.1.0

But in a newly created app, I don't use any of this dependancies 
<details>
<summary>pubspec.yaml</summary>

```

name: test_icons
description: A new Flutter project.

publish_to: 'none' # Remove this line if you wish to publish to pub.dev

version: 1.0.0+1

environment:
  sdk: ">=2.17.6 <3.0.5"

dependencies:
  flutter:
    sdk: flutter

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^2.0.0

flutter:

  uses-material-design: true

```

</details>

I've tried 
- `flutter clean` `flutter pub get` and also 
- `flutter pub cache repair`
but it didn't work.

**Comment by danagbemava-nc** [comment]
Hi @michaelprimez, 

> Unfortunately, is hard to tell with certainty.
I did 2 things before this problem appeared:

> 1. flutter upgrade and

From what version did you upgrade? 

Also, can you check on either the `beta` or `master` channels to see whether or not you still experience this issue?

**Comment by michaelprimez** [comment]
Problem solved.
In my case, I did 2 things:
1. I've installed the rosetta using the command `$ sudo softwareupdate --install-rosetta --agree-to-license`, taken from this [link](https://docs.flutter.dev/get-started/install/macos) and second and most important
2. I've update `Dart`  2.17.1 -> 2.17.6

Step 1, didn't have any effect as the icons continue to be wrong.

After step 2, I performed
`flutter clean`, `flutter pub cache repair` and then `flutter pub get` and everything is working as before.
